### PR TITLE
[cmake] Test that NDEBUG is disabled if asserts are requested

### PR DESCRIPTION
**Pull request by stwunsch**
Should we put this in as a canary?

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -435,3 +435,9 @@ if (NOT cxxmodules)
     )
   endif()
 endif()
+
+#--canary tests------------------------------------------------------------------------------------
+if(asserts AND NOT MSVC)
+  ROOT_EXECUTABLE(testCompilerFlags testCompilerFlags.cxx LIBRARIES Core)
```
Comment (Axel-Naumann) [inline test/CMakeLists.txt:441]: Can we rename this to be more specific, e.g. "checkAssertsNDEBUG"?

Comment (stwunsch) [reply test/CMakeLists.txt:441]: Yep, that's a good idea!

Comment (stwunsch) [reply test/CMakeLists.txt]: Done!

**Review by Axel-Naumann — APPROVED** [review]

**Comment by stwunsch** [comment]
Argh damn, forgot to squash :see_no_evil:

**Pull request merged**